### PR TITLE
Bug 1665052 - upgrade linux kernel for docker-worker images

### PR DESCRIPTION
**Pull request by djmitche**
(and docker-worker 37.1.0)

### CODE REVIEW

**Comment by djmitche** [comment]
I'll drop the docker-worker version bump from this due to #3521.

File: files/var/kernel/README.md [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
-  deb-src http://archive.ubuntu.com/ubuntu bionic-updates main
+deb-src http://archive.ubuntu.com/ubuntu bionic main
+deb-src http://archive.ubuntu.com/ubuntu bionic-updates main
 ```
-* `sudo apt-get build-dep linux linux-image-$(uname -r)`
-* `sudo apt-get install libncurses-dev flex bison openssl libssl-dev dkms libelf-dev libudev-dev libpci-dev libiberty-dev autoconf`
-* `apt-get source linux-image-$(uname -r)`
-* `cd linux-image*`
+* `sudo apt-get build-dep linux linux-image-$rev`
+* `sudo apt-get install libncurses-dev flex bison openssl libssl-dev dkms libelf-dev libudev-dev libpci-dev libiberty-dev autoconf fakeroot gcc-aarch64-linux-gnu`
```
Comment (petemoore) [inline files/var/kernel/README.md:13]: What is the cross compiler gcc toolchain needed for? (I see only amd64 debian packages in /files/var/kernel)

**Review by petemoore — APPROVED** [review]
It looks great, although I haven't understood how we build the arm64 variants for the arm64 docker-worker image sets (e.g. `a1.*` AWS instance types). Could you add a short explanation in the README about how we build for both architectures?

I've taken a look at worker pools on both community cluster and firefox-ci cluster, but couldn't see any `a1.*` instance types, so am now wondering if we no longer run arm64 workers in EC2? I thought we had migrated them from packet.net to EC2, but maybe we didn't?

**Comment by djmitche** [comment]
I don't know anything about arm64 workers.  Were those documented somewhere?  There weren't arm64 kernels in the repo when I started.

The `menuconfig` step fails without the cross compiler installed, even if I select "N" to configure arm64.  But it doesn't seem to *build* that architecture.

**Comment by djmitche** [comment]
I don't see anything in community-tc-config's `config/imagesets.yml` about amd64, either

**Comment by petemoore** [comment]
> I don't know anything about arm64 workers. Were those documented somewhere? There weren't arm64 kernels in the repo when I started.

I'm not sure - maybe @ccoper knows?

> The `menuconfig` step fails without the cross compiler installed, even if I select "N" to configure arm64. But it doesn't seem to _build_ that architecture.

Would it be ok to add a comment above the installation of `gcc-aarch64-linux-gnu` to explain that it is needed for `menuconfig` but that we don't _build_ for aarch64?

**Comment by petemoore** [comment]
> I don't see anything in community-tc-config's `config/imagesets.yml` about amd64, either

Do you mean arm64? No worries, if so, probably we're not using it.

**Comment by djmitche** [comment]
> Do you mean arm64?

I do!  Those are so similar :(

**Comment by ccooper** [comment]
> > I don't know anything about arm64 workers. Were those documented somewhere? There weren't arm64 kernels in the repo when I started.
> 
> I'm not sure - maybe @ccooper knows?

The existing nss ARM workers are here: https://firefox-ci-tc.services.mozilla.com/provisioners/localprovisioner/worker-types/nss-aarch64

Sadly this didn't get setup as a properly-managed pool. That's something we should change if this is a platform we'd want to offer more widely.

**Pull request merged**